File: LeetCode/Python/0001_Two_Sum.py
# 1. Two Sum
# https://leetcode.com/problems/two-sum/


class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        d = {key: value for value, key in enumerate(nums)}
        for i in range(len(nums)):
            value = target - nums[i]
            if value in d and d[value] != i:
                return [d[value], i]

File: LeetCode/JavaScript/0001-Two-Sum.js
// 1. Two Sum
// https://leetcode.com/problems/two-sum/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function (nums, target) {
  let rep = {};
  for (let i = 0; i < nums.length; i++) {
    const rest = target - nums[i];
    if (rep[rest] >= 0) {
      return [i, rep[rest]];
    }
    rep[nums[i]] = i;
  }
};
